typescript/web/src/components/labelling-tool/view-tool-bar/full-screen-tool.tsx

```diff
@@ -0,0 +1,127 @@
+import { RefObject, useState } from "react";
+import {
+  IconButton,
+  chakra,
+  Tooltip,
+  useColorModeValue as mode,
+} from "@chakra-ui/react";
+import { RiFullscreenLine, RiFullscreenExitLine } from "react-icons/ri";
+
+import { isInWindowScope } from "../../../utils/detect-scope";
+
+const FullscreenIcon = chakra(RiFullscreenLine);
+const FullscreenExitIcon = chakra(RiFullscreenExitLine);
+
+// For Full screen related stuff,
+// See https://github.com/openlayers/openlayers/blob/v6.7.0/src/ol/control/FullScreen.js
+
+/**
+ * @return {boolean} Fullscreen is supported by the current platform.
+ */
+function isFullScreenSupported() {
+  if (!isInWindowScope) {
+    // Assume it is supported for most users.
+    return true;
+  }
+  const { body } = document;
+  return !!(
+    body.webkitRequestFullscreen ||
+    (body.msRequestFullscreen && document.msFullscreenEnabled) ||
+    (body.requestFullscreen && document.fullscreenEnabled)
+  );
+}
+
+// /**
+//  * @return {boolean} Element is currently in fullscreen.
+//  */
+// function isFullScreen() {
+//   return !!(
+//     document.webkitIsFullScreen ||
+//     document.msFullscreenElement ||
+//     document.fullscreenElement
+//   );
+// }
+
+/**
+ * Request to fullscreen an element.
+ * @param {HTMLElement} element Element to request fullscreen
+ */
+function requestFullScreen(element: HTMLElement) {
+  if (element.requestFullscreen) {
+    element.requestFullscreen();
+  } else if (element.msRequestFullscreen) {
+    element.msRequestFullscreen();
+  } else if (element.webkitRequestFullscreen) {
+    element.webkitRequestFullscreen();
+  }
+}
+
+/**
+ * Request to fullscreen an element with keyboard input.
+ * @param {HTMLElement} element Element to request fullscreen
+ */
+function requestFullScreenWithKeys(element: HTMLElement) {
+  if (element.webkitRequestFullscreen) {
+    element.webkitRequestFullscreen();
+  } else {
+    requestFullScreen(element);
+  }
+}
```

Comment:
There's no keyboard shortcut for enabling fullscreen right? Plus `element.webkitRequestFullscreen()` is already called in `requestFullScreen`, is this really needed?

Reply:
All these functions are just a copy paste from openlayers code as said in the Comment on top. I can customize it if needed but I liked the idea of just keeping it as is

Reply:
But yes I can add a keyboard shortcut